### DISCUSSION TITLE
Historical Forecast Data

### DISCUSSION THREAD

**Discussion by mathnole** [open-meteo/open-meteo#468] [General]
First, thank you for this amazing API. 

I want to make sure that I understand what data is available in the API. Ideally I would like to obtain historical forecast data rather than actual (observed) weather data. However, I realize that this is difficult to provide given the data storage requirements. I am mainly looking at the Forecast API and the Historical API.

Forecast API: If a request is made to this API for a past date, do I receive the forecast for that date or the observed data? If it is a forecast, what was the time when it was archived? For example, if I request temperature data for 2023-09-30 do I get the forecast from some time on 2023-09-29?

Historical API: I believe this is based off of re-analysis of historical observed data. Is this correct?

Finally, I saw in another thread that you have archived forecasts back to some time in 2022. Is it possible to obtain that data either through the API or some other method?

Thanks!

**Comment by ckanaar** [comment]
Would like to know this as well, as the documentation is not very specific about this. In the Historical Weather API, it is stated that: 

"You can access past weather data dating back to 1940. However, there is a 5-day delay in the data. If you want information for the most recent days, you can use the [forecast API](https://open-meteo.com/en/docs) and adjust the Past Days setting."

So if we want information from the most recent days, we should refer to the Forecast API. However, in the Forecast API it is stated that:

"If you're interested in past weather data, you can use the Past Days feature to access archived forecasts."

So this refers to forecasts I believe, not to actual observed weather data. However, does this mean that in the Historical Weather API we are also retrieving past forecasts instead of past observed weather (for recent years?).

**Comment by ckanaar** [comment]
Follow up question with regards to the following statement:

"If you're interested in past weather data, you can use the Past Days feature to access archived forecasts."

Which archived forecasts are these exactly? With this I mean what is the date at which the forecast was created? For example if I'd select an archived forcast from 01-01-2023, at what day was the forecast created? 31-12-2022, 30-12-2022, or even before that?

**Reply by patrick-zippenfenig** [reply]
The weather forecast API is continuously integrating the latest weather model updates into a continuous time series. Because weather models update every 1, 3 or 6 hours, the resulting time-series is very close to the actual observed weather conditions. The largest remaining errors are gridding bias (1-10 km grid-cells) and inaccuracies in assimilated weather data. It should be relatively accurate, but is only available the for recent months.

The historical weather API uses the same principle, however with much more past data. It is based on weather reanalysis data with coarser resolution (25-10 km) but with up to 80 years of data. If you want to analyse trends in weather data (climate change, risk), this kind of data should be used.

In the future, I would like to integrate more datasets like satellite observations to better compare different weather data sources. Depending on the use case, some dataset might be better than another. There is no "one size fits all" unfortunately.
